### PR TITLE
Add default value to client functions that have optional form data or query params

### LINKED ISSUE #146: Error when calling LemmyHttp.getPosts() with no parameters
Error happens because `encodeGetParams()` calls `Object.entries()` on undefined. 

Solution is to just give the parameter a default value of `{}`

```
/node_modules/.pnpm/lemmy-js-client@0.18.0-rc.2/node_modules/lemmy-js-client/dist/http.js:815
    return Object.entries(p)
                  ^

TypeError: Cannot convert undefined or null to object
    at Function.entries (<anonymous>)
    at encodeGetParams (/node_modules/.pnpm/lemmy-js-client@0.18.0-rc.2/node_modules/lemmy-js-client/dist/http.js:815:19)
    at LemmyHttp.<anonymous> (/node_modules/.pnpm/lemmy-js-client@0.18.0-rc.2/node_modules/lemmy-js-client/dist/http.js:788:136)
    at Generator.next (<anonymous>)
    at /node_modules/.pnpm/lemmy-js-client@0.18.0-rc.2/node_modules/lemmy-js-client/dist/http.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (/node_modules/.pnpm/lemmy-js-client@0.18.0-rc.2/node_modules/lemmy-js-client/dist/http.js:4:12)
    at LemmyHttp._LemmyHttp_wrapper (/node_modules/.pnpm/lemmy-js-client@0.18.0-rc.2/node_modules/lemmy-js-client/dist/http.js:785:12)
    at LemmyHttp.getPosts (/node_modules/.pnpm/lemmy-js-client@0.18.0-rc.2/node_modules/lemmy-js-client/dist/http.js:275:92)
    at file:///index.mjs:5:26
```

### PR DESCRIPTION
**Pull request by cinderisles**
Fixes #146

### CODE REVIEW

**Review by dessalines — COMMENTED** [review]
I don't know about this one, because not all routes allow empty bodies. Did you test what happens to tooling for those routes?

**Comment by cinderisles** [comment]
tbh I didn't do much testing. im not using typescript, so i'll only see an error at runtime. this is my code

```js
// index.mjs
import { LemmyHttp } from 'lemmy-js-client'

const client = new LemmyHttp('http://localhost:1236')

console.log(await client.getPosts())
```

if the error is expected, it might be nice to throw one that's more informative. something that would suggest that the parameter for `getPosts()` is required

**Comment by cinderisles** [comment]
you could argue the error it throws is informative enough, yeah. but if i could save a few seconds when forgetting a parameter then great

File: src/http.ts [hunk 1]
```diff
@@ -1335,7 +1335,9 @@ export class LemmyHttp {
   }
 }
 
-function encodeGetParams<BodyType extends object>(p: BodyType): string {
+function encodeGetParams<BodyType extends object>(
```
[the remark this answers is not in the document]
Comment (cinderisles) [reply src/http.ts]: good idea, that will make the types more accurate. I think I got all of them in the new commit

**Review by SleeplessOne1917 — APPROVED** [review]

**Comment by cinderisles** [comment]
rebased to fix conflicts

**Pull request merged**